KinectIPD/KinectIPD/HoloDevice.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using System.Net;
+using System.Windows.Forms;
+using Microsoft.Tools.WindowsDevicePortal;
+
+namespace KinectIPD
+{
+    public class HoloDevice
+    {
+        public DevicePortal Portal;
+        public string DeviceName { get; set; }
+        public bool IsSaving { get; private set; }
+        public virtual bool IsValid
+        {
+            get
+            {
+                return !(
+                    string.IsNullOrWhiteSpace(DeviceName) ||
+                    string.IsNullOrWhiteSpace(ipAddress) ||
+                    string.IsNullOrWhiteSpace(username) ||
+                    string.IsNullOrWhiteSpace(password)
+                    );
+            }
+        }
+        protected string ipAddress = "";
+        protected string username = "";
+        protected string password = "";
+        protected bool autoDetectIP = false;
+
+        public HoloDevice()
+        {
+        }
+
+        public HoloDevice(string deviceName, string ipaddress, string username, string password)
+        {
+            DeviceName = deviceName;
+            ipAddress = ipaddress;
+            this.username = username;
+            this.password = password;
+        }
+
+        public virtual void ResetDevice()
+        {
+            DeviceName = "";
+            ipAddress = "";
+            username = "";
+            password = "";
+        }
+
+        public virtual bool Save()
```

Comment:
NIT: I wonder if this should be done using interfaces. 
Sorry without class descriptions it's hard to tell the intent.